models/document-generation.ts

```diff
@@ -0,0 +1,426 @@
+  NO_SOONER_THAN = "NO_SOONER_THAN",
+}
+
+export enum AddressType {
+  FOREIGN = "FOREIGN",
+  MILITARY = "MILITARY",
+  US = "US",
+}
+
+export enum Classification {
+  U = "U", // Unclassified
+  S = "S", // Secret
+  TS = "TS", // Top Secret
+}
+
+export enum ImpactLevel {
+  IL2 = "IL2",
+  IL4 = "IL4",
+  IL5 = "IL5",
+  IL6 = "IL6",
+}
+
+export enum InstanceLocation {
+  CSP = "CSP",
+  HYBRID = "HYBRID",
+  ON_PREMISE = "ON_PREMISE",
+}
+
+export enum TargetCspName {
+  CSP_A = "CSP_A",
+  CSP_B = "CSP_B",
+  CSP_C = "CSP_C",
+  CSP_D = "CSP_D",
+}
+
+export enum PricingModel {
+  ON_DEMAND = "ON_DEMAND",
+  PAY_AS_YOU_GO = "PAY_AS_YOU_GO",
+  RESERVED = "RESERVED",
+}
+
+export enum StorageUnit {
+  GB = "GB",
+  TB = "TB",
+}
+
+export enum ServiceOfferingGroup {
+  ADVISORY = "ADVISORY",
+  APPLICATIONS = "APPLICATIONS",
+  COMPUTE = "COMPUTE",
+  DATABASE = "DATABASE",
+  DEVELOPER_TOOLS = "DEVELOPER_TOOLS",
+  EDGE_COMPUTING = "EDGE_COMPUTING",
+  GENERAL_XAAS = "GENERAL_XAAS",
+  IOT = "IOT",
+  MACHINE_LEARNING = "MACHINE_LEARNING",
+  NETWORKING = "NETWORKING",
+  SECURITY = "SECURITY",
+  TRAINING = "TRAINING",
+}
+
+export interface IAward {
+  contract_award_type: AwardType;
+  effective_date: string;
+  modification_order: number | undefined;
+}
+
+export interface IContractInformation {
+  contract_information: string;
+  current_contract_exists: boolean;
+  contract_expiration_date: string;
+  incumbent_contractor_name: string;
+  previous_task_order_number: string;
+}
+
+export interface IPeriod {
+  period_type: PeriodType;
+  period_unit_count: number;
+  period_unit: PeriodUnit | string;
+  option_order: number;
+}
+
+export interface IClassificationLevel {
+  classification_level: Classification;
+  impact_level: ImpactLevel;
+}
+
+export interface IEnvironmentInstance {
+  instance_name: string;
+  classification_level: IClassificationLevel;
+  instance_location: InstanceLocation;
+  csp_region: TargetCspName;
+  performance_tier: string;
+  pricing_model: PricingModel;
+  pricing_model_expiration: string;
+  operating_system_licensing: string;
+  number_of_vCPUs: number;
+  storage_type: string;
+  storage_amount: number;
+  storage_unit: StorageUnit;
+  memory_amount: number;
+  memory_unit: StorageUnit;
+  data_egress_monthly_amount: number;
+  data_egress_monthly_unit: StorageUnit;
+}
+
+export interface ICurrentEnvironment {
+  current_environment_exists: boolean;
+  environment_instances: IEnvironmentInstance[];
+  additional_info: string;
+}
+
+export interface IClassificationInstance {
+  usage_description: string;
+  selected_periods: IPeriod[];
+  classification_level: IClassificationLevel;
+  need_for_entire_task_order_duration: boolean;
+}
+
+export interface IServiceOffering {
+  name: string;
+  service_offering_group: ServiceOfferingGroup;
+}
+
+export interface ISelectedServiceOffering {
+  service_offering: IServiceOffering;
+  classification_instances: IClassificationInstance[];
```

Comment:
Should this be nullable?

Reply:
I'm assuming this and the one below moot since our discussion, but if not let me know. Also, I think we would want to make sure these are not null since there should be classifications at the very least.